pubsub/observer.go

```diff
@@ -35,6 +35,12 @@ func (so *observer) isActive() bool {
 }
 
 func (so *observer) close() {
+	defer func() {
+		if err := recover(); err != nil {
+			// catch "close of closed channel"
```

Comment:
lets add error log here

Reply:
Added, as the logger wasn't used in all components I had to add it in multiple places.
It make sense to refactor `logger` in the future to be sort of a singleton instead of propagate it across our code-line